tests/cpp/unittests/testSplines.cpp

```diff
@@ -0,0 +1,498 @@
+        {2.0/3, {0.0,       0.0,      0.0}},
+        {0.75,  {-1.07812,  0.179688, 0.1875}},
+        {1.00,  {-6.0,      1.0,      3.0}},
+    };
+    test_spline_sensitivities(spline, expectations, 1e-5, 1e-6);
+    ASSERT_THROW(spline.get_sensitivity(-0.05, 0), AmiException);
+    ASSERT_THROW(spline.get_sensitivity( 1.05, 1), AmiException);
+}
+
+TEST(Splines, SplineNonUniformSensitivity)
+{
+    HermiteSpline spline({ 0.0, 0.1, 0.5, 1.0 },
+                         { 2.5, 3.25, 1.0, 4.5 },
+                         {},
+                         SplineBoundaryCondition::given,
+                         SplineBoundaryCondition::given,
+                         SplineExtrapolation::noExtrapolation,
+                         SplineExtrapolation::noExtrapolation,
+                         true,   // node_derivative_by_FD
+                         false,  // equidistant_spacing
+                         false); // logarithmic_parametrization
+    int n_params = 3;
+    std::vector<double> dvaluesdp = {
+         3.0,  1.0,  0.0,
+         0.0,  0.0,  5.0,
+         0.0,  0.0,  0.0,
+        -6.0,  1.0,  3.0
+    };
+    auto dslopesdp = std::vector<double>(spline.n_nodes() * n_params);
+    spline.compute_coefficients();
+    spline.compute_coefficients_sensi(n_params, 0, dvaluesdp, dslopesdp);
+    std::vector<std::tuple<double, std::vector<double>>> expectations = {
+        // t, expected values of sensitivities
+        {0.00, { 3.0,     1.0,     0.0}},
+        {0.05, { 1.3125,  0.4375,  2.89062}},
+        {0.10, { 0.0,     0.0,     5.0}},
+        {0.30, {-0.45,   -0.3,     3.6}},
+        {0.50, { 0.0,     0.0,     0.0}},
+        {0.75, {-2.625,   0.4375,  0.921875}},
+        {1.00, {-6.0,     1.0,     3.0}},
+    };
+    test_spline_sensitivities(spline, expectations, 1e-5, 1e-6);
+    ASSERT_THROW(spline.get_sensitivity(-0.05, 0), AmiException);
+    ASSERT_THROW(spline.get_sensitivity( 1.05, 1), AmiException);
+}
+
+TEST(Splines, SplineExplicitSensitivity)
+{
+    HermiteSpline spline({ 0.0, 1.0 },
+                         { 2.5, 3.25, 1.0, 4.5 },
+                         { 13.625, 7.5, 1.1585290151921035, 1.0 },
+                         SplineBoundaryCondition::given,
+                         SplineBoundaryCondition::given,
+                         SplineExtrapolation::noExtrapolation,
+                         SplineExtrapolation::noExtrapolation,
+                         false,  // node_derivative_by_FD
+                         true,   // equidistant_spacing
+                         false); // logarithmic_parametrization
+    int n_params = 3;
+    std::vector<double> dvaluesdp = {
+         3.0,  1.0,  0.0,
+         0.0,  0.0,  5.0,
+         0.0,  0.0,  0.0,
+        -6.0,  1.0,  3.0
+    };
+    std::vector<double> dslopesdp = {
+         0.0,  0.0,      18.75,
+         0.0,  1.0,       3.0,
+         4.0, -0.540302,  0.0,
+         0.0,  0.0,       0.0,
+    };
+    std::vector<double>(spline.n_nodes() * n_params);
```

Comment:
This is just constructing a vector and discarding it immediately. Supposed to be used somewhere?

Reply:
Just a leftover from copying and editing the previous test. Thanks for catching it!